### ISSUE TITLE
In my Debian 12 system PyRadio won't start

### ISSUE THREAD

**Issue by thegreatyellow67** [coderholic/pyradio#210]
Hi, I've just installed PyRadio in my SpiralLinux KDE Plasma 5.27.5 distribution (which is based on Debian 12 Bookworm). I followed the entire installation procedure for Debian 12 step by step, installing pipx. Running pyradio gives me this error:

Error: Module "dateutil" not found!

Please install the above module and try again.

Debian based distros:
sudo apt install python3-dateutil
.......
If everything else fails, try:
python -m pip install dateutil
or
python3 -m pip install dateutil
or even
python -m pip install --user dateutil

As message suggested, I also have installed python3-dateutil, but pyradio refuses to start. How can I solve it? I love pyradio, I would be able to use it on my freshly installed system. Please help me!
Thanks a lot!

**Comment by s-n-g** [comment]
Have you followed the [Fully isolated installation - Debian](https://github.com/coderholic/pyradio/blob/master/docs/linux-pipx.md#fully-isolated-installation---debian) or the [System dependent installation - Debian](https://github.com/coderholic/pyradio/blob/master/docs/linux-pipx.md#system-dependent-installation---debian) procedure?

**Comment by thegreatyellow67** [comment]
I've followed the Fully isolated one.

**Comment by s-n-g** [comment]
I just followed the installation process on my debian:

```
$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

And then
```
python3 install.py -i

                     _____       _____           _ _
                    |  __ \     |  __ \         | (_)
                    | |__) |   _| |__) |__ _  __| |_  ___
                    |  ___/ | | |  _  // _` |/ _` | |/ _ \
                    | |   | |_| | | \ \ (_| | (_| | | (_) |
                    |_|    \__, |_|  \_\__,_|\__,_|_|\___/
                            __/ |
                           |___/

                               installation script
                                   running on

                   _____       _   _                    ____
                  |  __ \     | | | |                  |___ \
                  | |__) |   _| |_| |__   ___  _ __      __) |
                  |  ___/ | | | __| '_ \ / _ \| '_ \    |__ <
                  | |   | |_| | |_| | | | (_) | | | |   ___) |
                  |_|    \__, |\__|_| |_|\___/|_| |_|  |____/
                          __/ |
                         |___/


    
Installing PyRadio...
Using directory: "/home/spiros/.config/pyradio/data/.cache"
Old "/home/spiros/.config/pyradio/data/.cache" found. Deleting...
Downloading PyRadio source code...
  url: "https://github.com/coderholic/pyradio/archive/0.9.2.20.zip"
  filename: "/home/spiros/.config/pyradio/data/.cache/pyradio-0.9.2.20.zip"
  ** file found in cache!
Extracting RyRadio source code...
  installed package pyradio 0.9.2.20, installed using Python 3.11.2
  These apps are now globally available
    - pyradio
done! ✨ 🌟 ✨
  injected package rich into venv pyradio
done! ✨ 🌟 ✨
  injected package requests into venv pyradio
done! ✨ 🌟 ✨
  injected package dnspython into venv pyradio
done! ✨ 🌟 ✨
  injected package psutil into venv pyradio
done! ✨ 🌟 ✨
  injected package netifaces into venv pyradio
done! ✨ 🌟 ✨
  injected package python-dateutil into venv pyradio
done! ✨ 🌟 ✨
Installing Desktop file ...   


PyRadio succesfully installed!
Hope you have a lot of fun using it!
Cheers!
```

This is what it gives me. You can see the last module it installs is `python-dateutil`

Are you sure you have followed the procedure correctly?

I would suggest to 
```
python3 -m pipx list
rm -rf ~/.local/pipx/venvs/pyradio
```

and start over from the top of [Debian and derivatives](https://github.com/coderholic/pyradio/blob/master/docs/linux-pipx.md#debian-and-derivatives)

**Comment by thegreatyellow67** [comment]
The result of python3 -m pipx list is:

venvs are in /home/tgy-tutorials/.local/pipx/venvs
apps are exposed on your $PATH at /home/tgy-tutorials/.local/bin
   package pyradio 0.9.2.20, installed using Python 3.11.2
    - pyradio

I'll try to rm -rf ~/.local/pipx/venvs/pyradio and re-do the installation procedure.

**Comment by thegreatyellow67** [comment]
No, I've yet installed those packages:

python3-full è già alla versione più recente (3.11.2-1+b1).
python3-pip è già alla versione più recente (23.0.1+dfsg-1).
python3-rich è già alla versione più recente (13.3.1-1).
python3-requests è già alla versione più recente (2.28.1+dfsg-1).
0 aggiornati, 0 installati, 0 da rimuovere e 0 non aggiornati.

**Comment by s-n-g** [comment]
So, this is the problem:
```
netifaces.c:1:10: fatal error: Python.h: File o directory non esistente
```

Python.h does not exist...
It should have been installed with `python3-full`...
Anyway...

Let's see the package that contains it
```
dpkg -S Python.h
libpython3.11-dev:amd64: /usr/include/python3.11/Python.h
```

It's `libpython3.11-dev`, so
```
sudo apt install libpython3.11-dev
```

and then, again

```
python3 install.py -i
```

**Comment by thegreatyellow67** [comment]
Yesss!! Yesss!! Yesss!!! Now it works!! Missed package!!!! Thanks a very very very lot!!! PyRadio is my favourite program!! :-)

**Comment by s-n-g** [comment]
I am happy it did!

I will update the instructions as well :wink: 
Closing...

**Comment by thegreatyellow67** [comment]
Hi, I would like to point out that today I had to update pyradio in my old LMDE5 installation (Debian 11), the library to install in all older versions of Debian 11 Bullseye is libpython3.9-dev, so:
sudo apt install libpython3.9-dev

**Comment by thegreatyellow67** [comment]
Yes, libpython3-all-dev exists. If I install this package the result of dpkg -s libpython3-all-dev is:

Package: libpython3-all-dev
Status: install ok installed
Priority: optional
Section: libdevel
Installed-Size: 6
Maintainer: Matthias Klose <doko@debian.org>
Architecture: amd64
Multi-Arch: same
Source: python3-defaults
Version: 3.9.2-3
Depends: libpython3-dev (= 3.9.2-3), libpython3.9-dev
Description: package depending on all supported Python 3 development packages
The package currently depends on libpython3.9-dev, in the
future, dependencies on jython (Python for a JVM) and ironpython (Python
for Mono) may be added.

If I searched for the library libpython3.11-dev it didn't exist (sure an upgrade that exists in Debian 12), so in LMDE5 I took the nearest as name, libpython3.9-dev. But if it's more correct to install the package libpython3-all-dev you were right to correct me, also for a better understanding of all users who install pyradio in Debian 11. Thank you again, also for your kindness and timeliness in responding to messages. Sorry for my not so good english, I hope the meaning is clear enough! Bye!
